### PR TITLE
fixing dispose method

### PR DESCRIPTION
**Pull request by tushargupta51**
#288
<a href='#crh-start'></a><a href='#crh-data-%7B%22processed%22%3A%20%5B%22https%3A//github.com/AzureAD/azure-activedirectory-identitymodel-extensions-for-dotnet/pull/295%23issuecomment-150059020%22%2C%20%22https%3A//github.com/AzureAD/azure-activedirectory-identitymodel-extensions-for-dotnet/pull/295%23issuecomment-151246152%22%2C%20%22https%3A//github.com/AzureAD/azure-activedirectory-identitymodel-extensions-for-dotnet/pull/295%23issuecomment-152245750%22%2C%20%22https%3A//github.com/AzureAD/azure-activedirectory-identitymodel-extensions-for-dotnet/pull/295%23discussion_r42832703%22%2C%20%22https%3A//github.com/AzureAD/azure-activedirectory-identitymodel-extensions-for-dotnet/pull/295%23discussion_r42911320%22%5D%2C%20%22comments%22%3A%20%7B%22General%20Comment%22%3A%20%7B%22html_url%22%3A%20%22https%3A//github.com/AzureAD/azure-activedirectory-identitymodel-extensions-for-dotnet/pull/295%23issuecomment-150059020%22%2C%20%22comments%22%3A%20%5B%7B%22body%22%3A%20%22%40brentschmaltz%20%22%2C%20%22created_at%22%3A%20%222015-10-22T00%3A11%3A57Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/803796%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/tushargupta51%22%7D%7D%2C%20%7B%22body%22%3A%20%22%40brentschmaltz%20about%20%5C%22call%20the%20base.Dispose%28disposing%29%3B..%5C%22%2C%20what%20I%20am%20referring%20here%20is%20that%20the%20method%20is%20abstract%20and%20it%20cannot%20be%20called%20from%20the%20derived%20class.%22%2C%20%22created_at%22%3A%20%222015-10-26T18%3A50%3A24Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/803796%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/tushargupta51%22%7D%7D%2C%20%7B%22body%22%3A%20%22%3Ashipit%3A%20however%2C%20we%20still%20have%20some%20additional%20work%20todo%2C%20since%20SecurityKey%20does%20not%20implement%20IDisposable%2C%20yet%20some%20of%20our%20keys%20are%20disposable.%22%2C%20%22created_at%22%3A%20%222015-10-29T16%3A49%3A22Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/3172421%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/brentschmaltz%22%7D%7D%5D%2C%20%22title%22%3A%20%22General%20Comment%22%7D%2C%20%22Pull%20479246107f1083dc63cd0e8222d0533f22a40184%20src/System.IdentityModel.Tokens/AsymmetricSignatureProvider.cs%2017%22%3A%20%7B%22html_url%22%3A%20%22https%3A//github.com/AzureAD/azure-activedirectory-identitymodel-extensions-for-dotnet/pull/295%23discussion_r42832703%22%2C%20%22comments%22%3A%20%5B%7B%22body%22%3A%20%22We%20need%20to%20do%20two%20things%3A%5Cr%5Cn1.%20We%20need%20a%20finalizer%5Cr%5Cn2.%20call%20the%20base.Dispose%28disposing%29%3B%20%3C---%20There%20is%20disagreement%20on%20what%20to%20pass%20%28true%20or%20disposing%29%20according%20to%3A%20https%3A//msdn.microsoft.com/en-us/library/fs2xkftw%28v%3Dvs.110%29.aspx%22%2C%20%22created_at%22%3A%20%222015-10-23T04%3A40%3A20Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/3172421%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/brentschmaltz%22%7D%7D%2C%20%7B%22body%22%3A%20%221.%20Why%20we%20need%20a%20finalizer%3F%20I%20don%27t%20think%20there%20are%20any%20unmanaged%20resources%20that%20we%20need%20to%20get%20%20rid%20of%3F%5Cr%5Cn2.%20base%20class%20-%20SignatureProvider%20is%20an%20abstract%20class.%20We%20don%27t%20need%20to%20call%20its%20dispose.%22%2C%20%22created_at%22%3A%20%222015-10-23T20%3A35%3A13Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/803796%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/tushargupta51%22%7D%7D%5D%2C%20%22title%22%3A%20%22File%3A%20src/System.IdentityModel.Tokens/AsymmetricSignatureProvider.cs%3AL466-488%22%7D%7D%2C%20%22approved%22%3A%20%7B%22https%3A//github.com/brentschmaltz%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/3172421%3Fv%3D3%22%7D%7D%7D'></a>
<a href='https://www.codereviewhub.com/'></a>

<a href='https://github.com/brentschmaltz'></a>

- [ ] <a href='#crh-comment-Pull 479246107f1083dc63cd0e8222d0533f22a40184 src/System.IdentityModel.Tokens/AsymmetricSignatureProvider.cs 17'></a> &nbsp;<b><a href='https://github.com/AzureAD/azure-activedirectory-identitymodel-extensions-for-dotnet/pull/295#discussion_r42832703'>File: src/System.IdentityModel.Tokens/AsymmetricSignatureProvider.cs:L466-488</a></b>
- <a href='https://github.com/brentschmaltz'></a> We need to do two things:
1. We need a finalizer
2. call the base.Dispose(disposing); <--- There is disagreement on what to pass (true or disposing) according to: https://msdn.microsoft.com/en-us/library/fs2xkftw(v=vs.110).aspx
- <a href='https://github.com/tushargupta51'></a> 1. Why we need a finalizer? I don't think there are any unmanaged resources that we need to get  rid of?
2. base class - SignatureProvider is an abstract class. We don't need to call its dispose.
- [x] <a href='#crh-comment-General Comment'></a> &nbsp;<b><a href='https://github.com/AzureAD/azure-activedirectory-identitymodel-extensions-for-dotnet/pull/295#issuecomment-150059020'>General Comment</a></b>
- <a href='https://github.com/tushargupta51'></a> @brentschmaltz
- <a href='https://github.com/tushargupta51'></a> @brentschmaltz about "call the base.Dispose(disposing);..", what I am referring here is that the method is abstract and it cannot be called from the derived class.
- <a href='https://github.com/brentschmaltz'></a> :shipit: however, we still have some additional work todo, since SecurityKey does not implement IDisposable, yet some of our keys are disposable.

<a href='https://www.codereviewhub.com/AzureAD/azure-activedirectory-identitymodel-extensions-for-dotnet/pull/295?mark_as_completed=1'></a>&nbsp;<a href='https://www.codereviewhub.com/AzureAD/azure-activedirectory-identitymodel-extensions-for-dotnet/pull/295?approve=1'></a>&nbsp;<a href='https://www.codereviewhub.com/AzureAD/azure-activedirectory-identitymodel-extensions-for-dotnet/pull/295?approve=0'></a>&nbsp;<a href='https://github.com/AzureAD/azure-activedirectory-identitymodel-extensions-for-dotnet/pull/295'></a>
<a href='#crh-end'></a>

### CODE REVIEW

**Comment by brandwe** [comment]
This pull request is now managed with CodeReviewHub. <a href='https://github.com/AzureAD/azure-activedirectory-identitymodel-extensions-for-dotnet/pull/295'>Refresh</a>.<a href='#crh-update-none'></a>

**Comment by tushargupta51** [comment]
@brentschmaltz

File: src/System.IdentityModel.Tokens/AsymmetricSignatureProvider.cs [hunk 1]
```diff
@@ -464,9 +466,22 @@ protected override void Dispose(bool disposing)
         {
```
Comment (brentschmaltz) [inline src/System.IdentityModel.Tokens/AsymmetricSignatureProvider.cs]: We need to do two things:
1. We need a finalizer
2. call the base.Dispose(disposing); <--- There is disagreement on what to pass according to: https://msdn.microsoft.com/en-us/library/fs2xkftw(v=vs.110).aspx

Comment (tushargupta51) [inline src/System.IdentityModel.Tokens/AsymmetricSignatureProvider.cs]: 1. Why we need a finalizer? I don't think there are any unmanaged resources that we need to get  rid of?
2. base class - SignatureProvider is an abstract class. We don't need to call its dispose.

**Comment by tushargupta51** [comment]
@brentschmaltz about "call the base.Dispose(disposing);..", what I am referring here is that the method is abstract and it cannot be called from the derived class.

**Comment by brentschmaltz** [comment]
:shipit: however, we still have some additional work todo, since SecurityKey does not implement IDisposable, yet some of our keys are disposable.

**Pull request merged**